File: Documentation/5.4/Samples/python/ClientApi/Session/HowTo/Clear.py
from examples_base import ExampleBase, Employee


class Foo:
    # region clear_1
    def clear(self) -> None: ...

    # endregion


class ClearExample(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_example1(self):
        with self.embedded_server.get_document_store("Clear") as store:
            with store.open_session() as session:
                # region clear_2
                session.store(Employee(first_name="John", last_name="Doe"))

                session.advanced.clear()

                session.save_changes()  # nothing will happen
                # endregion

File: Documentation/5.4/Samples/php/ClientApi/session/HowTo/Clear.php
<?php

use RavenDB\Documents\DocumentStore;

interface FooInterface {
    # region clear_1
    public function clear(): void;
    # endregion
}

class Clear
{
    public function example(): void
    {
        $store = new DocumentStore();
        try {
            $session = $store->openSession();
            try {
                # region clear_2
                $employee = new Employee();
                $employee->setFirstName("John");
                $employee->setLastName("Doe");
                $session->store($employee);

                $session->advanced()->clear();

                $session->saveChanges(); // nothing will hapen
                # endregion
            } finally {
                $session->close();
            }
        } finally {
            $store->close();
        }
    }
}

class Employee
{
    private ?string $firstName = null;
    private ?string $lastName = null;

    public function getFirstName(): ?string
    {
        return $this->firstName;
    }

    public function setFirstName(?string $firstName): void
    {
        $this->firstName = $firstName;
    }

    public function getLastName(): ?string
    {
        return $this->lastName;
    }

    public function setLastName(?string $lastName): void
    {
        $this->lastName = $lastName;
    }
}
